ietf-network-inventory.yang

```diff
@@ -136,79 +131,165 @@ module ietf-network-inventory {
     description
       "Attributes applicable to network elements.";
     leaf hardware-rev {
+      config false;
       type string;
       description
         "The vendor-specific hardware revision string for the NE.";
     }
     leaf software-rev {
+      config false;
       type string;
       description
         "The vendor-specific software revision string for the NE.";
     }
     leaf mfg-name {
+      config false;
       type string;
       description "The name of the manufacturer of this NE";
     }
     leaf mfg-date {
+      config false;
       type yang:date-and-time;
       description "The date of manufacturing of the NE.";
     }
-    leaf part-number {
-      type string;
-      description
-        "The vendor-specific model name identifier string associated
-         with this NE.  The preferred value is the customer-visible 
-         part number, which may be printed on the NE itself.";
-    }
     leaf serial-number {
+      config false;
       type string;
       description
         "The vendor-specific serial number string for the NE";
     }
     leaf product-name {
+      config false;
       type string;
       description
         "indicates the vendor-spefic device type infomation.";
     }
   }
   
-  container network-elements {
-    description
-      "The top-level container for the list of network elements 
-      within the network.";
-    list network-element {
-      key ne-id;
+  grouping component-specific-info-grouping {
+    description 
+      "Provide component-specific attributes for different 
+      components.";
+    choice component-class {
+      config false;
```

Comment:
Unless we agree that this model can only be used as RO, it is possible that some component-specific attribute might be RW. It is suggested not to preclude this option:

```suggestion
```

Reply:
Now this structure is configable. But i have a concern that the change from RW to RO will be a NBC.

Reply:
Please note that the NBC change requirement applies only to YANG modules published as IETF RFCs. NBC changes are allowed between different versions of I-Ds.

The issues related to the level of configuration is required for the inventory model are still open: https://github.com/ietf-ivy-wg/network-inventory-yang/issues?q=is%3Aissue+is%3Aopen+label%3Aconfig